test/engine-luatest/gh_5501_errinj_index_iterator_new_test.lua

```diff
@@ -0,0 +1,36 @@
+local t = require('luatest')
+local g = t.group('gh-5501', {{engine = 'memtx'},
+                              {engine = 'vinyl'}})
+
+g.before_all(function(cg)
+    local server = require('test.luatest_helpers.server')
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:stop()
+    cg.server = nil
+end)
+
+-- Check that box.commit() doesn't fail like:
+-- "txn_commit_stmt: Assertion `txn->in_sub_stmt > 0' failed"
+-- or "Transaction is active at return from function"
+-- or "Can not commit transaction in a nested statement".
+g.test_errinj = function(cg)
+    local engine = cg.params.engine
+
+    cg.server:exec(function(engine)
+        local s = box.schema.space.create('test', {engine = engine})
+        s:create_index('pk')
+
+        -- This s:select{0} will fail because of the error injection.
+        s:on_replace(function() pcall(s.select, s, 0) end)
```

Comment:
Let's add test cases for other functions, as well. We have just a just few of them and all of them seem to be affected to the very same issue and can be tested in a similar fashion: `min()`, `max()`, `get()`, `count()`, `pairs()`.